### ISSUE TITLE
docker build on main fails (rust:1.57-slim-bookworm image not found)

### ISSUE THREAD

**Issue by ripcurlz** [influxdata/influxdb_iox#8212]
__Steps to reproduce:__

1. run docker build command as stated in README.md on main branch
2. results in error `docker.io/library/rust:1.57-slim-bookworm: not found`

quick search on docker hub shows that tag "1.57-slim-bookworm" is not there (anymore): 

https://hub.docker.com/_/rust/tags?page=1&name=1.57-slim

__Logs:__

```
PS C:\influxdb_iox> set DOCKER_BUILDKIT=1; docker build -t influx:v3 .
[+] Building 13.8s (6/6) FINISHED
 => [internal] load build definition from Dockerfile                                                                                                                          0.0s
 => => transferring dockerfile: 2.29kB                                                                                                                                        0.0s
 => [internal] load .dockerignore                                                                                                                                             0.0s
 => => transferring context: 2B                                                                                                                                               0.0s
 => resolve image config for docker.io/docker/dockerfile:1.2                                                                                                                  1.9s
 => docker-image://docker.io/docker/dockerfile:1.2@sha256:e2a8561e419ab1ba6b2fe6cbdf49fd92b95912df1cf7d313c3e2230a333fdbcc                                                   10.4s
 => => resolve docker.io/docker/dockerfile:1.2@sha256:e2a8561e419ab1ba6b2fe6cbdf49fd92b95912df1cf7d313c3e2230a333fdbcc                                                        0.0s
 => => sha256:e2a8561e419ab1ba6b2fe6cbdf49fd92b95912df1cf7d313c3e2230a333fdbcc 1.69kB / 1.69kB                                                                                0.0s
 => => sha256:e3ee2e6b536452d876b1c5aa12db9bca51b8f52b2505178cae6d13e33daeed2b 528B / 528B                                                                                    0.0s
 => => sha256:86e43bba076d67c1a890cbc07813806b11eca53843dc643202d939b986c8c332 1.21kB / 1.21kB                                                                                0.0s
 => => sha256:3cc8e449ce9f6e0752ede8f50a7334bf0c7b2d24d76da2ffae7aa6a729dd1da4 9.64MB / 9.64MB                                                                               10.2s
 => => extracting sha256:3cc8e449ce9f6e0752ede8f50a7334bf0c7b2d24d76da2ffae7aa6a729dd1da4                                                                                     0.1s
 => CANCELED [internal] load metadata for docker.io/library/debian:bookworm-slim                                                                                              1.2s
 => ERROR [internal] load metadata for docker.io/library/rust:1.57-slim-bookworm                                                                                              1.2s
------
 > [internal] load metadata for docker.io/library/rust:1.57-slim-bookworm:
------
failed to solve with frontend dockerfile.v0: failed to solve with frontend gateway.v0: rpc error: code = Unknown desc = docker.io/library/rust:1.57-slim-bookworm: not found
```

**Comment by crepererum** [comment]
We currently don't keep the dockerfile rust version and the one specified in `rust-toolchain.toml` in sync. You can fix that by passing `--arg RUST_VERSION=1.71` to `docker build`.

**Comment by ripcurlz** [comment]
works for me with `--build-arg RUST_VERSION=1.71`. thank you!
